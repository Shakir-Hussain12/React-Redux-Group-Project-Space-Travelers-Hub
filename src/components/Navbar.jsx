import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import Container from 'react-bootstrap/Container';

import logo from '../Images/planet.png';

import styles from './Navbar.module.css';

const links = [
  { path: '/', text: 'Rockets' },
  { path: '/missions', text: 'Missions' },
  { path: 'myprofile', text: 'My Profile' },
];

function Navbar({ menu, handleMenu }) {
  return (
    <header className={`${styles.header} border-bottom`}>
      <Container className={styles.nav}>
        <NavLink to={links[0].path} className={styles.logo} style={{ border: 0, color: '#1b1b1b' }}>
          <img src={logo} alt="logo" />
          <span className={styles.logoName}>Space Traveler&apos;s Hub</span>
        </NavLink>
        <ul className={`${styles.links} d-none d-md-flex`}>
          {links.map((link) => (
            <li key={link.text}>
              <NavLink
                to={link.path}
                className={({ isActive }) => (isActive ? 'active' : undefined)}
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
        {menu && (
          <ul className={`${styles.mobileNav} d-md-none py-4`}>
            <Container>
              {links.map((link) => (
                <li key={link.text}>
                  <NavLink
                    to={link.path}
                    className={`${({ isActive }) => (isActive ? 'active' : undefined)} ${
                      styles.option
                    } border p-2 mb-2 rounded`}
                    onClick={handleMenu}
                  >
                    {link.text}
                  </NavLink>
                </li>
              ))}
            </Container>
          </ul>
        )}
        <button
          className={`${styles.hamburger} ${menu ? styles.open : styles.close} d-md-none`}
          type="button"
          onClick={handleMenu}
        >
          <span className={`${styles.stroke} ${styles.stroke1}`} />
          <span className={`${styles.stroke} ${styles.stroke2}`} />
          <span className={`${styles.stroke} ${styles.stroke3}`} />
        </button>
      </Container>
    </header>
  );
}

Navbar.propTypes = {
  menu: PropTypes.bool.isRequired,
  handleMenu: PropTypes.func.isRequired,
};

export default Navbar;
