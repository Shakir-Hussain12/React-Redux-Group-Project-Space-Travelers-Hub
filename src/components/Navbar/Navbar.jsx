import { NavLink } from 'react-router-dom';
import Container from 'react-bootstrap/Container';

import styles from './Navbar.module.css';

const links = [
  { path: '/', text: 'Missions' },
  { path: 'myprofile', text: 'My Profile' },
];

function Navbar() {
  return (
    <header>
      <Container className={styles.nav}>
        <span>
          <NavLink
            to={links[0].path}
            className={styles.logo}
            style={{ border: 0, color: '#1b1b1b' }}
          >
            Space Traveler&apos;s Hub
          </NavLink>
        </span>
        <ul className={styles.links}>
          {links.map((link) => (
            <li key={link.text}>
              <NavLink
                to={link.path}
                className={({ isActive }) => (isActive ? 'active' : undefined)}
              >
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </Container>
    </header>
  );
}

export default Navbar;
