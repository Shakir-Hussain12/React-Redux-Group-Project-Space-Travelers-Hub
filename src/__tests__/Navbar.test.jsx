import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import Navbar from '../components/Navbar';

describe('Navbar tests', () => {
  test('should render Navbar', () => {
    const header = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );

    expect(header).toMatchSnapshot();
  });
});
