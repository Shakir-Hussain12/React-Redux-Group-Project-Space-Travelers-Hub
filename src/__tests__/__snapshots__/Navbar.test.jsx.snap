// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar tests > should render Navbar 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="_header_39d119 border-bottom"
      >
        <div
          class="_nav_39d119 container"
        >
          <a
            aria-current="page"
            class="_logo_39d119 active"
            href="/"
            style="border: 0px; color: rgb(27, 27, 27);"
          >
            <img
              alt="logo"
              src="/src/Images/planet.png"
            />
            <span
              class="_logoName_39d119"
            >
              Space Traveler's Hub
            </span>
          </a>
          <ul
            class="_links_39d119"
          >
            <li>
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li>
              <a
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                href="/myprofile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="_header_39d119 border-bottom"
    >
      <div
        class="_nav_39d119 container"
      >
        <a
          aria-current="page"
          class="_logo_39d119 active"
          href="/"
          style="border: 0px; color: rgb(27, 27, 27);"
        >
          <img
            alt="logo"
            src="/src/Images/planet.png"
          />
          <span
            class="_logoName_39d119"
          >
            Space Traveler's Hub
          </span>
        </a>
        <ul
          class="_links_39d119"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li>
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              href="/myprofile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
